src/main/java/org/prebid/server/cache/CacheService.java

```diff
@@ -219,6 +242,39 @@ private static String createUuid(CacheObject cacheObject) {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Creates a map with bids as a key and {@link CacheIdInfo} as a value from obtained UUIDs.
+     */
+    private static <T> Map<T, CacheIdInfo> toResultMap(List<T> bids, List<T> videoBids, List<String> uuids) {
+        final Map<T, CacheIdInfo> result = new HashMap<>(uuids.size());
+
+        for (int i = 0; i < bids.size(); i++) {
+            final T bid = bids.get(i);
+            if (!result.containsKey(bid)) {
```

Comment:
Is it possible that bid is already in result map at this step?

Reply:
Yeah, you're right. I can skip it safely.